x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/UpdateInferenceModelAction.java

```diff
@@ -0,0 +1,278 @@
+         * If any fields in the body are not on the allow list, this function will throw an exception.
+         */
+        public Settings getContentAsSettings() {
+            if (settings == null) { // settings is deterministic on content, so we only need to compute it once
+                Map<String, Object> unvalidatedMap = XContentHelper.convertToMap(content, false, contentType).v2();
+                Map<String, Object> serviceSettings = new HashMap<>();
+                Map<String, Object> taskSettings = new HashMap<>();
+                TaskType taskType = null;
+
+                if (unvalidatedMap.isEmpty()) {
+                    throw new ElasticsearchStatusException("Request body is empty", RestStatus.BAD_REQUEST);
+                }
+
+                if (unvalidatedMap.containsKey("task_type")) {
+                    if (unvalidatedMap.get("task_type") instanceof String taskTypeString) {
+                        taskType = TaskType.fromStringOrStatusException(taskTypeString);
+                    } else {
+                        throw new ElasticsearchStatusException(
+                            "Failed to parse [task_type] in update request [{}]",
+                            RestStatus.INTERNAL_SERVER_ERROR,
+                            unvalidatedMap.toString()
+                        );
+                    }
+                    unvalidatedMap.remove("task_type");
+                }
+
+                if (unvalidatedMap.containsKey(SERVICE_SETTINGS)) {
+                    if (unvalidatedMap.get(SERVICE_SETTINGS) instanceof Map<?, ?> tempMap) {
+                        for (Map.Entry<?, ?> entry : (tempMap).entrySet()) {
+                            if (entry.getKey() instanceof String key) {
+                                serviceSettings.put(key, entry.getValue());
+                            } else {
+                                throw new ElasticsearchStatusException(
+                                    "Failed to parse update request [{}]",
+                                    RestStatus.INTERNAL_SERVER_ERROR,
+                                    unvalidatedMap.toString()
+                                );
+                            }
+                        }
+                        unvalidatedMap.remove(SERVICE_SETTINGS);
+                    } else {
+                        throw new ElasticsearchStatusException(
+                            "Unable to parse service settings in the request [{}]",
+                            RestStatus.BAD_REQUEST,
+                            unvalidatedMap.toString()
+                        );
+                    }
+                }
+
+                if (unvalidatedMap.containsKey(TASK_SETTINGS)) {
+                    if (unvalidatedMap.get(TASK_SETTINGS) instanceof Map<?, ?> tempMap) {
+                        for (Map.Entry<?, ?> entry : (tempMap).entrySet()) {
+                            if (entry.getKey() instanceof String key) {
```

Comment:
I had to change this line and the similar line above to only use instanceOf on the key